lib/charliecloud.py

```diff
@@ -2032,3 +2074,10 @@ def tree_terminals_cat(tree, tname):
 def unlink(path, *args, **kwargs):
    "Error-checking wrapper for os.unlink()."
    ossafe(os.unlink, "can't unlink: %s" % path, path)
+
+def user():
+   "Return the current username; exit with error if it can't be obtained."
+   try:
+      return os.environ["USER"]
+   except KeyError:
+      FATAL("can't get username: $USER not set")
```

Comment:
Are there any scenarios where USER might not be set and we would want to use something like `whoami` as a fallback?

Reply:
I've wondered about this. I think all of Charliecloud is trash if `$USER` isn't set though, so it's a bigger problem than right here.